fighthealthinsurance/utils.py

```diff
@@ -46,31 +48,50 @@ def is_convertible_to_int(s):
         return False
 
 
-def send_fallback_email(subject: str, template_name: str, context, to_email: str):
-    if to_email.endswith("-fake@fighthealthinsurance.com"):
-        return
-    # First, render the plain text content if present
-    text_content = render_to_string(
-        f"emails/{template_name}.txt",
-        context=context,
-    )
+def send_fallback_email(subject: str, template_name: str, context, to_email: str) -> bool:
+    """Send an email with both text and HTML fallback, ensuring proper error handling."""
+    try:
+        if not to_email:
+            raise ValidationError("Recipient email is required.")
+
+        # Render email templates
+        text_content = render_to_string(f"emails/{template_name}.txt", context=context).strip()
+        html_content = render_to_string(f"emails/{template_name}.html", context=context).strip()
+
+        # Ensure there's at least one valid content type
+        if not text_content and not html_content:
+            raise ValidationError("Both text and HTML templates are empty.")
+
+        # Create email instance
+        msg = EmailMultiAlternatives(
+            subject,
+            text_content or "This email has no text content.",
+            settings.EMAIL_HOST_USER,
+            [to_email],
+        )
+
+        # Attach HTML content if available
+        if html_content:
+            msg.attach_alternative(html_content, "text/html")
+
+        msg.send()
+        return True  # Email sent successfully
+
+    except BadHeaderError:
+        logger.error("Invalid email header detected.")
+        return False
 
-    # Secondly, render the HTML content if present
-    html_content = render_to_string(
-        f"emails/{template_name}.html",
-        context=context,
-    )
-    # Then, create a multipart email instance.
-    msg = EmailMultiAlternatives(
-        subject,
-        text_content,
-        settings.EMAIL_HOST_USER,
-        [to_email],
-    )
+    except SMTPException as e:
+        logger.error(f"SMTP error while sending email to {to_email}: {e}")
+        return False
+
+    except ImproperlyConfigured as e:
+        logger.error(f"Email settings misconfigured: {e}")
+        return False
```

Comment:
I think we've had this conversation a few times, we want to throw an exception on failure here.